File: python/rna-transcription/rna_transcription.py
def to_rna(dna_strand):
    dna2rna = {
        "G" : "C",
        "C" : "G",
        "T" : "A",
        "A" : "U"
    }
    
    return ''.join(dna2rna[c] for c in dna_strand)

File: typescript/rna-transcription/rna-transcription.ts
class Transcriptor {
    static readonly RNA2DNA = new Map<string,string>()
        .set('C','G')
        .set('G','C')
        .set('A','U')
        .set('T','A');

    toRna(dna: string): string {
        let res = [];
        for(const c of dna){
            if(Transcriptor.RNA2DNA.has(c)) {
                res.push(Transcriptor.RNA2DNA.get(c))
            } else {
                throw 'Invalid input DNA.';
            }
        }
        return res.join('');
    }
}

export default Transcriptor
